static/app/views/dashboardsV2/widgetCard/metricsWidgetQueries.tsx

```diff
@@ -28,8 +36,155 @@ type State = {
 };
 
 class MetricsWidgetQueries extends React.Component<Props, State> {
+  state: State = {
+    loading: true,
+    queryFetchID: undefined,
+    errorMessage: undefined,
+    timeseriesResults: undefined,
+    rawResults: undefined,
+    tableResults: undefined,
+  };
+
+  componentDidMount() {
+    this._isMounted = true;
+    this.fetchData();
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    // TODO(dam): Put more sophisticated refetch logic in place
+    const {selection, widget} = this.props;
+    if (
+      !isEqual(widget.displayType, prevProps.widget.displayType) ||
```

Comment:
looks like we're checking for display type 2x here and in like 59. We should also be firing the query if widget queries (fields, y axis, conditions, orderbys etc) change.